controllers/configuration_controller_test.go

```diff
@@ -200,6 +200,155 @@ func TestInitTFConfigurationMetaWithJobNodeSelector(t *testing.T) {
 	assert.Equal(t, meta.JobNodeSelector, map[string]string{"ssd": "true"})
 }
 
+func TestPrepareTFVariables(t *testing.T) {
+	variable, _ := json.Marshal(map[string]interface{}{
+		"Test": "abc123",
+	})
+	configuration := v1beta2.Configuration{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "default",
+			Name:      "abc",
+		},
+		Spec: v1beta2.ConfigurationSpec{
+			ProviderReference: &crossplane.Reference{
+				Name:      "default",
+				Namespace: "default",
+			},
+			HCL: "test",
+			Variable: &runtime.RawExtension{
+				Raw: variable,
+			},
+		},
+	}
+	meta := &TFConfigurationMeta{
+		ProviderVariable: map[string]interface{}{
+			"PrjID": "test123",
+		},
+		ProviderReference: &crossplane.Reference{
+			Name:      "default",
+			Namespace: "default",
+		},
+		Credentials: map[string]string{
+			"key":    "testkey",
+			"secret": "testsecret",
+		},
+	}
+	err := meta.prepareTFVariables(&configuration)
+	assert.Nil(t, err)
+	assert.Equal(t, meta.VariableSecretData["PrjID"], []byte("test123"))
+	assert.Equal(t, meta.VariableSecretData["TF_VAR_Test"], []byte("abc123"))
+	assert.Equal(t, meta.VariableSecretData["key"], []byte("testkey"))
+	envExistPrjID := false
+	envExistTest := false
+	envExistKey := false
+	for _, e := range meta.Envs {
+		if e.Name == "PrjID" {
+			envExistPrjID = true
+		}
+		if e.Name == "TF_VAR_Test" {
+			envExistTest = true
+		}
+		if e.Name == "key" {
+			envExistKey = true
+		}
+	}
+	if !envExistPrjID {
+		assert.Nil(t, fmt.Errorf("meta.Envs not exist PrjID"))
+	}
+	if !envExistTest {
+		assert.Nil(t, fmt.Errorf("meta.Envs not exist Test"))
+	}
+	if !envExistKey {
+		assert.Nil(t, fmt.Errorf("meta.Envs not exist key"))
+	}
```

Comment:
It could be more concise. Consider using map and for-loop to reduce the repetition.